ivy.el

```diff
@@ -1963,14 +1963,14 @@ This is useful for recursive `ivy-read'."
          (let ((w (length (number-to-string
                            (length ivy--all-candidates))))
                (s (copy-sequence ivy-count-format)))
-           (string-match "%d" s)
+           (string-match-p "%d" s)
            (match-end 0)
-           (string-match "%d" s (match-end 0))
+           (string-match-p "%d" s (match-end 0))
            (setq s (replace-match (format "%%-%dd" w) nil nil s))
```

Comment:
Doesn't `replace-match` use `match-data`? I think this will introduce a bug. Can you confirm?

Reply:
Not only `replace-match` uses `match-data`, but also the `match-end` passed to the second `string-match` from the first `string-match`. So neither of the two `string-match`es should be changed, at least at first glance.

Reply:
Yes, my changes are wrong here.